fe1-web/src/core/components/Icon.tsx

```diff
@@ -0,0 +1,129 @@
+import Ionicons from '@expo/vector-icons/Ionicons';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import { ExtendType } from 'core/types';
+
+type IoniconNames = keyof typeof Ionicons['glyphMap'];
+
+type Ionicon = {
+  iconName: IoniconNames;
+  IconFamily: typeof Ionicons;
+};
+
+const iconNameMap = {
```

Comment:
I guess it depends :) It is definitely complicated I agree. The reason was to still enable the IDE to check the "name" prop we are passing.

`expo/vector-icons` exports multiple icon families, `Ionicons` is just one of them and certain icons are not present in it. (Not important for this PR but for the next one since there we also use a few material ui icons)

Now my goal was to have a uniform interface no matter what icon family is used (implementation detail). This was possible with each Icon being a separate component and is now possible with this map. The problem is that the Icon component somehow needs to know which icon family the icon is from, i.e. `Ionicons` or `MaterialIcons`. Previously this was just defined in each file and now there is a map from "our name" to a tuple of icon family and the name this icon has inside this icon family. This way the Icon component can look up this map and use the correct combination of icon family and name. We could omit the `IonIcon` type but I thought it would help preventing mistakes since now the linter will tell you if there is an invalid combination of icon family and icon name. (`IoniconNames` is a union type of all valid icon names).

And because we have this map from our names to the tuples, the linter can also check whether a valid name was passed to the Icon component (Whose map entry points to a valid tuple).

So basically a lot of work so that the linter can check everything. But if you think this is not necessary and people using the Icon component will know what they are doing, we can remove all of this and have a simpler file.

In any case I agree that I should add comments to this file is it is definitely not trivial 😅

Reply:
Mhmm ok, I understand why you did this now. Maybe another solution would be to create different icon components for each family? Imagine having a `Ionicon` component and a `MaterialIcon` component, so that you can omit the family and create 2 separated collections of names? This might not scale well for a lot of Icon families, but as I understand we're using two of them at the moment so it shouldn't be an issue. What do you think?

Reply:
That would also be an option yes. This would almost result in the same interface `expo/vector-icons` provides, just with a restricted set of names. I am not sure whether this is better but if you think this is better I am happy to change it :)

Reply:
Let's go with your solution for now, we can still modify it later if needed.